mautic/mautic
# REST API for forms allows to send certain fields with HTML tags

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [No]
| New feature/enhancement? (use the a.x branch)      | [Yes]
| Deprecations?                          | [No]
| BC breaks? (use the c.x branch)        | [No]
| Automated tests included?              | [No] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:

A REST API request to `{{domain}}/api/forms/{{id}}/edit` allows to send an html-formatted form text field and an html-formatted email message body (actions). 

```
{
    "isPublished": true,
    "id": 1,
    "name": "Test",
    "fields": [
        {
             ...
            "properties": {
                "text": "{
    "isPublished": true,
    "id": 1,
    "name": "Test",
    "fields": [
        {
             ...
            "properties": {
                "text": "<ul><li>Test 1</li><li>Test 2</li><li>Test 3</li><li>Test 4</li><li>Test 5</li></ul>"
            },
            ...
        }
    ],
    "actions": [
        {
             ....
            "properties": {
                ....
                "message": "Hello, <br style=\"box-sizing: border-box;\" /><br style=\"box-sizing: border-box;\" /> this is a test message.<br style=\"box-sizing: border-box;\" /><br style=\"box-sizing: border-box;\" /><a href=\"https://en.wikipedia.org/wiki/%22Hello,_World!%22_program\" style=\"user-select: auto; background-color: transparent; box-sizing: border-box; color: rgb(40, 62, 76); text-decoration: none;\">Test link</a>",
                .... 
            }
        }
    ],
    ...
}"

```

E.g., 
saved without HTML tags
![image](https://user-images.githubusercontent.com/96085911/152299629-8b5b28c0-867d-4f56-9139-0cdc9893c0fb.png)
![image](https://user-images.githubusercontent.com/96085911/152301276-959d9938-f260-47a2-aac9-7210c455d05c.png)

saved with HTML tags
![image](https://user-images.githubusercontent.com/96085911/152299688-83c58ee5-a460-4fe4-8dc4-9672ac3ac33c.png)
![image](https://user-images.githubusercontent.com/96085911/152301344-7a6d22fd-7a9d-4827-859e-7037f52a9cae.png)




#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Send POST request to {{domain}}/api/forms/new or PUT to {{domain}}/api/forms/{{id}}/edit
3. Check result, e.g by the url "{{domain}}/s/forms/edit/{{id}}"




<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/10820"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
